src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,44 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private User user = new User();
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (isUserValid(user)) {
+            storageDao.add(user);
+        }
+        return user;
+    }
+
+    public boolean isUserValid(User user) {
+        if (user == null) {
+            throw new RuntimeException("User can’t be null");
+        }
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        checkAge(user.getAge());
+        return true;
+    }
+
+    private void checkLogin(String login) {
+        if (login == null || storageDao.get(login) != null) {
+            throw new RuntimeException("User with same login is exist");
+        }
+    }
```

Comment:
so if `login == null` we will have the exception with message "User with same login is exist"